src/cli/telemetry/telemetry.ts

```diff
@@ -0,0 +1,293 @@
+import { tryFn, isInteractive, hasDebug, readJson, hasVerbose } from './helpers';
+import { CompilerBuildResults, CompilerSystem, Config, Logger, PackageJsonData } from 'src/declarations';
+import { parseFlags } from '../parse-flags';
+import { loadCoreCompiler } from '../load-compiler';
+import { readConfig, updateConfig } from '../ionic-config';
+import { isObject } from '@utils';
+import { getCompilerSystem } from '../state/stencil-cli-config';
+
+export interface CommandMetricData {
+  // app_id: string;
+  command: string;
+  arguments: string;
+  options: string;
+  duration: number;
+  error: string | null;
+  node_version: string;
+  // os: string;
+}
+
+export interface Metric<N extends string, D> {
```

Comment:
For the interfaces in this file, consider adding a JSDoc that gives a high level description of the interface. That way, when someone mouses over the interface, an editor's intellisense can provide that description to the dev, rather than making them do archaeology to figure out what objects of this shape/interface are